xblock/fields.py

```diff
@@ -646,6 +650,24 @@ def is_set_on(self, xblock):
 
     def __hash__(self):
         return hash(self.name)
+    
+    @classmethod
+    def Query(cls, field_name):
+        """
+        Return a instance of Query which allow ad-hoc queries to access shared fields. 
+        """
+        return Query(field_name)
```

Comment:
Thoughts on type of cls and type of querying field: if we instantiate Query as an independent object instead of attaching it to Field, we can stop worry about mismatch types and Query uses whatever the querying field defines.